Public/Src/Cache/ContentStore/Distributed/NuCache/ClusterState.cs

```diff
@@ -130,5 +130,10 @@ public bool TryResolveMachineId(MachineLocation machineLocation, out MachineId m
         {
             return _idByLocationMap.TryGetValue(machineLocation, out machineId);
         }
+
+        /// <summary>
+        /// Gets all locations that are currently known.
+        /// </summary>
+        public MachineLocation[] GetKnownLocations() => _idByLocationMap.Keys.ToArray();

```

Comment:
Yeah, it looks like this does create a new array on every call. To avoid this, use a read-only wrapper or maybe just create a GetRandomLocation API that operates on existing location map.